inference/core/workflows/execution_engine/introspection/blocks_loader.py

```diff
@@ -82,16 +92,39 @@ def get_manifest_type_identifiers(
     raise PluginInterfaceError(
         public_message="`type` property for block is required to be `Literal` "
         "defining at least one unique value to identify block in JSON "
-        f"definitions. Block `{block_identifier}` loaded from `{block_source} "
+        f"definitions. Block `{block_identifier}` loaded from `{block_source}` "
         f"does not fit that requirement.",
-        context="workflow_compilation | blocks_loading",
+        context="blocks_loading",
     )
 
 
-def load_workflow_blocks() -> List[BlockSpecification]:
+def load_workflow_blocks(
+    execution_engine_version: Optional[Union[str, Version]] = None,
+) -> List[BlockSpecification]:
+    if isinstance(execution_engine_version, str):
+        try:
+            execution_engine_version = Version(execution_engine_version)
```

Comment:
here we are creating instance of Version but only if `execution_engine_version` is `str`, in any other case we pass something that is not instance of `Version` further down - maybe we should do something like this instead:

```python
    if not execution_engine_version:
        execution_engine_version = ""
    execution_engine_version = str(execution_engine_version)
    try:
        execution_engine_version = Version(execution_engine_version)
    (...)
```